woocommerce/woocommerce
# Treat the remote_logging feature as if it were a legacy feature.

Our feature management system can detect if a plugin is a WooCommerce extension and, for non-legacy features, it will potentially add warnings to the UI if those extensions have not declared themselves to be compatible. Sometimes this is undesirable, and I believe that to be true in the case of Remote Logging. For example, in this screenshot:
 
![remote-logging-warning](https://github.com/user-attachments/assets/2b8c3fb8-711e-4a60-bebd-1889d2655267)

For clarity: this is undesirable because I don't think we have any expectations whatsoever that WooCommerce extensions should have to declare themselves as being compatible with remote error logging.

In such cases, it makes sense to register the feature as being a legacy feature. Longer term, we'll probably want to change the name of the `is_legacy` property to something else, or else introduce a new property that makes things a little clearer. For now, I'm just trying to clear the problem to reduce user confusion.

### Testing instructions

1. Start with the current stable release (or with `trunk`, but without this change). 
2. Create a new plugin, with the following code (basically just a plugin header):

```php
<?php
/**
 * Plugin name: Test plugin
 * WC tested up to: 9.4
 */
```

3. Activate it. Also ensure that HPOS is enabled via your feature settings.
4. Now navigate to the plugin admin page and you should see a warning reading, *WooCommerce has detected that some of your active plugins are incompatible with currently enabled WooCommerce features. Please [review the details](https://example/wp-admin/plugins.php?plugin_status=incompatible_with_feature).*
5. Follow that link and you should see an inline notice next to our test plugin reading, *This plugin is incompatible with the enabled WooCommerce features 'Remote Logging' and 'High-Performance order storage', it shouldn't be activated. [Manage WooCommerce features](https://example/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features)*.
6. Now checkout this branch. The above should still basically be true (the warning should still be present) but it will no longer warn about *'Remote Logging'.*
7. Additionally, we should verify that remote logging still works. There are already tests in place to cover this, but one could also run through the steps listed in https://github.com/woocommerce/woocommerce/pull/50351 for additional confidence.
